packages/core/integration-tests/test/cache.js

```diff
@@ -6788,4 +6790,43 @@ describe('cache', function () {
     let res = await run(build.bundleGraph);
     assert.deepEqual(res, {default: 'foo'});
   });
+
+  it('invalidates correctly when switching from lazy to eager modes', async function () {
```

Comment:
I assume this was a failing test before your changes?

Reply:
yep, it was!